core/src/Kokkos_CopyViews.hpp

```diff
@@ -2041,10 +1477,365 @@ void KOKKOS_INLINE_FUNCTION local_deep_copy(
+                              iType, TeamMemberType> const& policy,
+                          ViewType const& view, std::index_sequence<Idx...>) {
+  if constexpr (UseSizeT) {
+    return Kokkos::ThreadVectorMDRange<
+        Kokkos::Rank<sizeof...(Idx), IterOrder, IterOrder>, TeamMemberType>(
+        policy.member, view.extent(Idx)...);
+  } else {
+    return Kokkos::ThreadVectorMDRange<
+        Kokkos::Rank<sizeof...(Idx), IterOrder, IterOrder>, TeamMemberType>(
+        policy.member, view.extent_int(Idx)...);
+  }
+}
+
+/** \brief  Deep copy with multidimensional views.  */
+template <class PolicyType, class Functor, class DataType, class... Properties>
+void KOKKOS_INLINE_FUNCTION md_local_deep_copy(
+    const PolicyType& policy, const Kokkos::View<DataType, Properties...>& dst,
+    const Functor& functor) {
+  constexpr std::size_t rank = ViewTraits<DataType, Properties...>::rank;
+
+  const Kokkos::Iterate iterate = Kokkos::Impl::get_iteration_order(dst);
+  if (dst.span() >= static_cast<size_t>(finite_max_v<int>)) {
+    if (iterate == Kokkos::Iterate::Right) {
+      Kokkos::parallel_for(
+          md_local_deep_copy_policy<Kokkos::Iterate::Right, true>(
+              policy, dst, std::make_index_sequence<rank>{}),
+          functor);
+    } else if (iterate == Kokkos::Iterate::Left) {
+      Kokkos::parallel_for(
+          md_local_deep_copy_policy<Kokkos::Iterate::Left, true>(
+              policy, dst, std::make_index_sequence<rank>{}),
+          functor);
+    } else {
+      Kokkos::parallel_for(
+          md_local_deep_copy_policy<Kokkos::Iterate::Default, true>(
+              policy, dst, std::make_index_sequence<rank>{}),
+          functor);
+    }
+  } else {
+    if (iterate == Kokkos::Iterate::Right) {
+      Kokkos::parallel_for(
+          md_local_deep_copy_policy<Kokkos::Iterate::Right, false>(
+              policy, dst, std::make_index_sequence<rank>{}),
+          functor);
+    } else if (iterate == Kokkos::Iterate::Left) {
+      Kokkos::parallel_for(
+          md_local_deep_copy_policy<Kokkos::Iterate::Left, false>(
+              policy, dst, std::make_index_sequence<rank>{}),
+          functor);
+    } else {
+      Kokkos::parallel_for(
+          md_local_deep_copy_policy<Kokkos::Iterate::Default, false>(
+              policy, dst, std::make_index_sequence<rank>{}),
+          functor);
+    }
+  }
+}
+
+//----------------------------------------------------------------------------
+template <class DestViewType, class SrcViewType>
+struct MDCopyFunctor {
+  DestViewType dst;
+  SrcViewType src;
+
+  KOKKOS_INLINE_FUNCTION MDCopyFunctor(DestViewType dst_, SrcViewType src_)
+      : dst(dst_), src(src_) {}
+
+  template <typename... Indices>
+  KOKKOS_INLINE_FUNCTION void operator()(const Indices... indices) const {
+    dst(indices...) = src(indices...);
+  }
+};
+
+template <class DestViewType, class ValueType>
+struct MDValueCopyFunctor {
+  DestViewType dst;
+  const ValueType value;
```

Comment:
```suggestion
  ValueType value;
```

I don't think it is that useful, given the other variables could also be marked `const`.